### PR TITLE
#430 e2e test fixes login and students

### PR DESCRIPTION
**Pull request by doritrieur**
some more tests on login and new student area. some fixes on classes area.

### CODE REVIEW

**Comment by eran1205** [comment]
Hi,

There are conflicts, can you fix?

File: tests/login.e2e.ts [hunk 1]
```diff
@@ -4,10 +4,15 @@ import { EyesDriver } from './eyes/eyes';
...
 const loginPage = new LoginPage();
 const eye = new EyesDriver();
+const password = 'Aa123456';
+const principleUser = 'principle';
+const principleName = 'MSW PRINCIPLE';
+const teacherUser = 'teacher';
 
-const assertLoginFailure = async (t) => {
+const assertOnLoginPage = async (t) => {
   const location = await t.eval(() => window.location);
```
Comment (dimkinv) [inline tests/login.e2e.ts]: please use `ClientFunction` intead
https://github.com/DevExpress/testcafe/blob/master/docs/articles/documentation/test-api/obtaining-data-from-the-client/examples-of-using-client-functions.md

File: tests/login.e2e.ts [hunk 2]
```diff
@@ -36,18 +41,18 @@ test('Successful login for teacher', async (t) => {
...
-  await assertLoginFailure(t);
+  await assertOnLoginPage(t);
 });
 
 test('Failed login - wrong credentials', async (t) => {
   // wrong password
-  await loginPage.login('teacher', 'wrong-password');
+  await loginPage.login(teacherUser, 'wrong-password');
   await t.expect(loginPage.wrongCredentialsErr.exists).ok();
-  await assertLoginFailure(t);
```
Comment (dimkinv) [inline tests/login.e2e.ts]: expect is chainable please use chains:
```
  await t
  .expect(loginPage.wrongCredentialsErr.exists)
  .ok()
  .expect(location.pathname).notContains('student')
  .expect(location.pathname).contains('login');
```

explicit is better than implicit
https://www.python.org/dev/peps/pep-0020/

File: tests/login.e2e.ts [hunk 3]
```diff
@@ -36,18 +41,18 @@ test('Successful login for teacher', async (t) => {
...
 test('Failed login for student', async (t) => {
   await loginPage.loginAsStudent();
-  await assertLoginFailure(t);
+  await assertOnLoginPage(t);
 });
 
 test('Failed login - wrong credentials', async (t) => {
   // wrong password
-  await loginPage.login('teacher', 'wrong-password');
+  await loginPage.login(teacherUser, 'wrong-password');
```
Comment (dimkinv) [inline tests/login.e2e.ts]: consider using `Role` factory for this:
https://devexpress.github.io/testcafe/documentation/test-api/authentication/user-roles.html

File: tests/students.e2e.ts [hunk 4]
```diff
@@ -24,75 +27,93 @@ test('new and existing student form error messages display correctly', async (t)
...
-    .ok()
-    .expect(studentPage.lastNameErr().exists)
-    .ok()
-    .expect(studentPage.usernameErr().exists)
-    .ok()
-    .expect(studentPage.classIdErr().exists)
-    .ok()
-    .expect(studentPage.passwordErr().exists)
-    .ok()
+    .expect(studentPage.firstNameErr.exists)
```
Comment (dimkinv) [inline tests/students.e2e.ts]: can you elaborate why this changed?

File: tests/login.e2e.ts (hunk 1 above)
Comment (doritrieur) [reply tests/login.e2e.ts]: implemented.

File: tests/login.e2e.ts (hunk 3 above)
Comment (doritrieur) [reply tests/login.e2e.ts]: did now know it. implementing.

**Pull request merged**